tests/Integration/PrestaShopBundle/Controller/Sell/Catalog/ProductControllerTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Sell\Catalog;
+
+use PrestaShop\PrestaShop\Core\Exception\TypeException;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Tests\Integration\PrestaShopBundle\Controller\GridControllerTestCase;
+use Tests\Integration\PrestaShopBundle\Controller\TestEntityDTO;
+
+class ProductControllerTest extends GridControllerTestCase
```

Comment:
I am sorry for not realizing sooner but isn't it weird that all the testing logic happens in the `setUp()` ?

The `setUp()` is supposed to prepare the environment for tests to happen, not to be the real test ? Also it means we can only have 1 test per GridControllerTestCase?

Reply:
The test happening here because I need unique entity with unique values so I can be sure that filters will return only value when used.

So I am creating this test entity during the setup and then I am asserting that we have all original entities + 1(the newly created) to ensure that tests are happening in correct enviroment.

I can throw exception instead of doing assertion, I don't know if there is a difference in this case.